### PR TITLE
[MM-53254] Fix make pluginapi

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost/pull/23345 moved the `public` directory under the new `public` go module. This broke the `pluginapi` make target, as it doesn't pick up the `//go:generate` directives any longer.

I suppose the reason is the newly added go module. This PR fixes the issue by running `go generate` from inside the `public` module.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53254

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]
LGTM, just a question below.

**Review by lieut-data — APPROVED** [review]
Good catch, @hanzei!

File: server/Makefile [hunk 1]
```diff
@@ -403,7 +403,7 @@ mmctl-mocks: ## Creates mocks for mmctl
 	$(GOBIN)/mockgen -destination=cmd/mmctl/mocks/client_mock.go -copyright_file=cmd/mmctl/mocks/copyright.txt -package=mocks github.com/mattermost/mattermost/server/v8/cmd/mmctl/client Client
 
 pluginapi: ## Generates api and hooks glue code for plugins
-	$(GO) generate $(GOFLAGS) ./public/plugin
+	-cd ./public && $(GO) generate $(GOFLAGS) ./plugin
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/Makefile]: The `-` ignores any non-zero exit code of that command - which is actually not what we want here. I did copy-paste it without thinking about it more. Fixed!

[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/Makefile]: TIL, thanks!

**Pull request merged**